src/js/requestHandlers.js

```diff
@@ -230,6 +235,85 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (storyContent, serverConfig) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        var blockFileName = "";
+
+        if (block.blockType === "image") {
+            blockFileName = block.imageUrl;
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        sjrk.storyTelling.server.deleteSingleFileRecoverable(fileToDelete, serverConfig);
+    });
+};
+
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName, serverConfig) {
+    try {
+        if (!fileName || fileName === "null" || fileName === "undefined")
+        {
+            throw "The filename is not valid: not truthy.";
+        }
+
+        if (sjrk.storyTelling.server.isValidUuidFileName(fileName)) {
+            throw "The filename is not valid: does not match UUID file name pattern.";
+        }
+
+        var fileWithFullPath = sjrk.storyTelling.server.getServerPathForFile(fileName, serverConfig.uploadedFilesHandlerPath);
+        fs.accessSync(fileWithFullPath, fs.constants.W_OK | fs.constants.R_OK);
+
+        return true; // it is a valid file name
+    } catch (err) {
+        return false; // it is not a valid file name
+    }
+};
+
+sjrk.storyTelling.server.isValidUuidFileName = function (fileName) {
+    var validUuid = /^\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/;
+    return !validUuid.test(path.basename(fileName));
```

Comment:
I don't think we want `path.basename` here.

Reply:
I misread the docs for path.basename() and didn't pass in the "ext" parameter which will remove the file extension from the rest of the filename. Would you suggest not using that function anyway?